Extension/test/unit/eventing.test.ts

```diff
@@ -0,0 +1,255 @@
+    }
+}
+
+// make an async constructor for the Something class
+// eslint-disable-next-line @typescript-eslint/naming-convention
+const AsyncSomething = Async(Something);
+
+describe('Event Emitters', () => {
+
+    beforeEach(() => {
+        reset();
+        // uncomment the following line to show debug messages in the console.
+        void on('debug', async (event: EventData) => { console.debug(event.text); });
+    });
+
+    it('try self-bound handlers', async () => {
+        const something = new Something();
+
+        // subscribe the object to its own events
+        subscribe(something);
+
+        // do something that will trigger the event
+        await something.init();
+
+        ok(something.initHandlerRan, 'The init handler should have cancelled the event');
+    });
+
+    it('try object handlers', async () => {
+        const something = new Something();
+
+        let count = 0;
+
+        const subscriber = {
+            'something/initialize': () => {
+                console.debug('in here !');
+                count++;
+                return Continue;
+            }
+        };
+
+        subscribe(subscriber, { bindAll: true });
+
+        await something.init();
+        strictEqual(count, 1, 'The init handler should have increased count by 1');
+
+        // create another, that wasn't existing before the subscription
+        const something2 = new Something();
+        await something2.init();
+        strictEqual(count, 2, 'The init handler should have increased count by 1 again');
+
+    });
+
+    it('ensure that `this` modifiers work correctly', async () => {
+        let countThisInit = 0;
+        let countAnyInit = 0;
+
+        const s1 = new Something();
+        const s2 = new Something();
+
+        const subscriber = {
+            // should only get called for events on the object it is bound to
+            'this initialize': () => {
+                console.debug('`this init` called');
+                countThisInit++;
+            },
+
+            // should get called for all events named 'init'
+            'initialize': () => {
+                console.debug('`init` called');
+                countAnyInit++;
+            }
+        };
+
+        // subscribe the handlers to events
+        subscribe(subscriber, { bindAll: true, eventSource: s1 });
+
+        await s1.init();
+        await s2.init();
+
+        strictEqual(countThisInit, 1, 'The `this` init handler should only get called on one of the instances');
+        strictEqual(countAnyInit, 2, 'The non-`this` init handler should get called for all instances');
+
+    });
+
+    it('show that `on` automatically assumes `this`', async () => {
+        let countThisInit = 0;
+
+        const s1 = new Something();
+        const s2 = new Something();
+
+        s1.on('initialize', () => {
+            console.debug('`this init` called');
+            countThisInit++;
+        });
+
+        // subscribe the handlers to events

```

Comment:
The comment says "subscribe the handlers to events" but I don't see any subscribe calls. Should the comment be removed or is there a call missing?

Reply:
comment should be above. fixed